src/components/policy/policy_external/src/cache_manager.cc

```diff
@@ -3121,13 +3126,23 @@ void CacheManager::BackgroundBackuper::threadMain() {
   LOG4CXX_AUTO_TRACE(logger_);
   sync_primitives::AutoLock lock(need_backup_lock_);
   while (!stop_flag_) {
+    backup_is_in_progress_.exchange(true);
     {
       sync_primitives::AutoUnlock need_backup_lock(need_backup_lock_);
       InternalBackup();
     }
+
     if (new_data_available_ || stop_flag_) {
```

Comment:
@AKalinich-Luxoft this case will exit while loop with `backup_is_in_progress_=true`. It it correct?

Reply:
@LuxoftAKutsan yes, looks like potentially some thread which is in `WaitForBackupIsDone()` might be stuck if backuper thread will be requested to stop. It's a good point. I moved `backup_is_in_progress_` setting before `continue` so now there will be a guarantee that `backup_is_in_progress_` will never be set to `true` if loop was exited by some reason.
See https://github.com/smartdevicelink/sdl_core/pull/3099/commits/32d697135fa021d2cf45366d4ebf0ac9e941accd